src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        checkPositionIndex(size);
+        Object[] obj = list.toArray();
+        int length = obj.length;
+        if (length == 0) {
+            return;
+        }
+        Node<T> prev;
+        prev = tail;
+        for (Object object : obj) {
+            T value = (T) object;
+            Node<T> newNode = new Node<>(prev, value, null);
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+        tail = prev;
+        size += length;
```

Comment:
you can do that, just try, in other case you can return to this version of your code

Reply:
> 

Okay.. I will try it {{{(>_<)}}}

Reply:
I did it...